packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -755,12 +755,46 @@ class ScrollableState extends State<Scrollable> with TickerProviderStateMixin, R
     );
   }
 
-  // Returns the delta that should result from applying [event] with axis and
-  // direction taken into account.
+  // Returns the delta that should result from applying [event] with axis,
+  // direction, and any modifier specified by the ScrollBehavior taken into
+  // account for horizontal scrollers.
   double _pointerSignalEventDelta(PointerScrollEvent event) {
-    double delta = widget.axis == Axis.horizontal
-      ? event.scrollDelta.dx
-      : event.scrollDelta.dy;
+    final TargetPlatform platform = _configuration.getPlatform(context);
+    final bool horizontal = widget.axis == Axis.horizontal;
+
+    final Set<LogicalKeyboardKey> pressed = HardwareKeyboard.instance.logicalKeysPressed;
+    final bool keyModifiersAreActive = pressed.any(_configuration.pointerAxisModifiers.contains);
+    pressed.removeAll(_configuration.pointerAxisModifiers);
+    // Ensure no other keys are currently pressed so as to not conflict with
+    // other shortcuts.
+    final bool axisIsFlipped = keyModifiersAreActive & pressed.isEmpty;
+    double delta = horizontal ? event.scrollDelta.dx : event.scrollDelta.dy;
+
+    if (horizontal & axisIsFlipped) {
+      switch(platform) {
+        case TargetPlatform.macOS:
+          // Mac already handles shift to flip the input axis, so if the
+          // modifier is currently LogicalKeyboardKey.shift (the default),
+          // ignore.
+        final bool alreadyHandled = HardwareKeyboard.instance.logicalKeysPressed.any(
+            <LogicalKeyboardKey>{
+              LogicalKeyboardKey.shiftLeft,
+              LogicalKeyboardKey.shiftRight,
+            }.contains);
+          if (alreadyHandled) {
+            break;
+          }
+          continue flip;
+        flip:
+        case TargetPlatform.android:
+        case TargetPlatform.fuchsia:
+        case TargetPlatform.iOS:
```

Comment:
does a mouse on iOS operate differently than on macOS?

Reply:
I think I have checked this, but I will check again and leave a comment. :)